### PR TITLE
Jenkins Slaves Configuration & SonarQube generator

### PR DESCRIPTION
**Pull request by parc-jason**
Fix for the Jenkins slaves, the slave images were never being used - the jenkins image was being deployed as a slave.  Changed the openshift templates and jenkins-overwrites to correctly deploy the slave image.  Added node to the slave images so they can be used to build and run tests for sonarqube (assuming a node project). 

Added a sonarqube "generator", included a sample jenkinsfile in the sonarqube README demonstrating how to use call sonarqube to run tests and publish results from a slave.

### CODE REVIEW

**Comment by patricksimonian** [comment]
can you reset your PR to not have the `create label.yml` commit? That commit belongs to a PR that is a work in progress :)

**Comment by patricksimonian** [comment]
@cvarjao what do you think of the jenkins image having node instead of using the `npmw` script?

**Comment by patricksimonian** [comment]
thanks for this contribution! We'll take a look at it and get back to you with any questions :) I'm wondering if this should be scheduled for a `0.1.0` release of the bcdk since it adds a major feature 🤔

**Comment by parc-jason** [comment]
No worries.
Just FYI, the SonarQube files were from [BCDevOps/sonarqube](https://github.com/BCDevOps/sonarqube/tree/bc80961d75eed66ec70ca022a6444963341fb39f) in May.  I see it has been bumped up, so could update the included yaml to use their latest?

**Comment by cvarjao** [comment]
@patricksimonian, I feel like each team could layer whatever utilities they need. I don't want to get in the business of managing Jenkins slave images for each technology again.

File: generators/jenkins/templates/.jenkins/openshift/build-slave.yaml [hunk 1]
```diff
@@ -51,13 +48,17 @@ objects:
...
-            rm -rf /tmp/sonnar-scanner-cli.zip && \
-            rm -rf /tmp/sonnar-scanner-cli && \
-            fix_permission '/opt/flywaydb' '/opt/liquibase' '/opt/sonnar-scanner'
+            curl -sSL -o /tmp/sonar-scanner-cli.zip https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-${SONAR_VERSION}-linux.zip && \
+            unzip /tmp/sonar-scanner-cli.zip -d /tmp/sonar-scanner-cli && \
+            mv /tmp/sonar-scanner-cli/sonar-scanner-${SONAR_VERSION}-linux /opt/sonar-scanner && \
+            ln -s /opt/sonar-scanner/bin/sonar-scanner /usr/local/bin && \
+            rm -rf /tmp/sonar-scanner-cli.zip && \
+            rm -rf /tmp/sonar-scanner-cli && \
+            curl -sSL https://nodejs.org/dist/${NODE_VERSION}/node-${NODE_VERSION}-linux-x64.tar.xz | tar -Jx -C /opt && \
```
Comment (patricksimonian) [inline generators/jenkins/templates/.jenkins/openshift/build-slave.yaml]: let's  not  include node as apart of the image. The BCDK  provides a npm wrapper  to utilize  inside of a jenkins file.  This is preferred over us managing changes to node within the  jenkins image.

**Comment by patricksimonian** [comment]
closing this pr as its gone stale

**Pull request closed without merging**